File: problems/week_2/readabilty/readability.go
package readability

import (
	"math"
	"strconv"
)

func Readability(text string) string {
	var length = len(text)

	var word_counter float64 = 0      // количество слов
	var letter_counter float64 = 0    // количество букв
	var sentences_counter float64 = 0 // количество предложений

	i := 0

	for i < length { // данный цикл необходим для подсчитывания букв, слов и предложений в тексте

		if (text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z') {

			letter_counter++ // отсчитывает буквы

		} else if text[i] == ' ' {

			word_counter++ // при помощи пробела отсчитывает слова

		} else if text[i] == '.' || text[i] == '!' || text[i] == '?' {

			sentences_counter++ // при помощи знаков препинания отсчитывает предложения

		}
		if length-1 == i && (text[i] == '.' || text[i] == '!' || text[i] == '?') {
			word_counter++ // при помощи данного условия отсчитывает последнее слово в тексте
		}
		i++
	}
	average_letter := (letter_counter / word_counter) * 100                  //  среднее количество букв на 100 слов в тексте
	average_sentences := (sentences_counter / word_counter) * 100            // среднее количество предложений в тексте
	var idx = (0.0588 * average_letter) - (0.296 * average_sentences) - 15.8 // Данная формула узнаёт к какому уровню сложности относится тексте
	if idx < 1 {
		return "Before Grade 1" // Если уровень сложности текста меньше первого класса
	} else if idx > 16 {
		return "Grade 16+" // если уровень сложности текста выше 16-го класса
	}
	var grade = int(math.Round(idx)) // Так как значение даётся с частным, его необходимо округлить для точной оценки класса

	class := "Grade " + strconv.Itoa(grade) // Строка, которая будет возвращать уровень класса

	return class
}

File: problems/week_6/readability_py/readability.py
import math


def main():
    text = get_string("Text:")
    text = text.upper()
    word = 0
    letter = 0
    sentence = 0
    for i in range(len(text)):
        if text[i] >= 'A' and text[i] < 'Z':
            letter += 1
        elif text[i] == ' ':
            word += 1
        elif text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentence += 1
        if i == len(text) - 1 and (text[i] == '.' or text[i] == '!' or text[i] == '?'):
            word += 1
    averageL = (float(letter) / word) * 100
    averageS = (float(sentence) / word) * 100
    idx = round((0.0588 * averageL) - (0.296 * averageS) - 15.8)
    if idx < 1:
        print("Before grade 1")
        return
    elif idx > 16:
        print("Grade 16+")
        return

    print("Grade:", idx)


def get_string(value):
    while True:
        try:
            return input(value)
        except:
            pass


main()
